Lib/test/test_ast/test_ast.py

```diff
@@ -3035,3 +3035,152 @@ def test_cli_file_input(self):
         self.assertEqual(expected.splitlines(),
                          res.out.decode("utf8").splitlines())
         self.assertEqual(res.rc, 0)
+
+
+class ASTOptimiziationTests(unittest.TestCase):
+    binop = {
+        "+": ast.Add(),
+        "-": ast.Sub(),
+        "*": ast.Mult(),
+        "/": ast.Div(),
+        "%": ast.Mod(),
+        "<<": ast.LShift(),
+        ">>": ast.RShift(),
+        "|": ast.BitOr(),
+        "^": ast.BitXor(),
+        "&": ast.BitAnd(),
+        "//": ast.FloorDiv(),
+        "**": ast.Pow(),
+    }
+
+    unaryop = {
+        "~": ast.Invert(),
+        "+": ast.UAdd(),
+        "-": ast.USub(),
+    }
+
+    def find_node(self, tree, target):
+        for node in ast.walk(tree):
+            if ast.compare(node, target):
+                return True
+        return False
+
+    def assert_ast(self, code, non_optimized_target, optimized_target):
```

Comment:
I think it would be better to use an approach that tests that the complete AST is what we expect, instead of only looking for subtrees. That would give more confidence that we're returning exactly what we'd expect.

To make creating the whole AST easier, you could add helper methods, e.g. one that takes a single expression and wraps it in `ast.Module(body=[ast.Expr(value=expr)])`.

Reply:
Thanks for the advice, Jelle.
I'm agree that this is more accurate way to do it.